src/statefulmodelinstance.cpp

```diff
@@ -104,12 +104,18 @@ Status StatefulModelInstance::reloadModel(const ModelConfig& config, const Dynam
     return StatusCode::OK;
 }
 
-void StatefulModelInstance::unloadModel(bool isPermanent, bool isError) {
+void StatefulModelInstance::retireModel(bool isPermanent) {
     std::lock_guard<std::recursive_mutex> loadingLock(loadingMutex);
-    if (isPermanent && autoCleanupEnabled) {
+    if (autoCleanupEnabled) {
         globalSequencesViewer->unregisterFromCleanup(getName(), getVersion());
     }
-    ModelInstance::unloadModel(isPermanent, isError);
+    ModelInstance::retireModel(isPermanent);
+    sequenceManager.reset();
```

Comment:
Why do we reset here? Don't we want to preserve sequences in case model will be loaded again? @mzegla

Reply:
Shouldn't this be as well in the if(isPermanent) section?

I think if we unload the model (not temporarily for reload) it may be ok, but this probably should happen inside retireModel?